backend/src/main/java/com/swappee/controller/privateapi/request/RequestPrivateController.java

```diff
@@ -1,239 +1,286 @@
+    }
+
+
+    /**
+     * Api to update a request to trade items
+     *
+     * @param requestDTO
+     * @return
+     */
+    @PutMapping("/{id}")
+    public ResponseEntity<ContentResult> updateRequest(@PathVariable Long id, @Valid @RequestBody RequestDTO requestDTO) {
+        logger.info("Start updateRequest - id: {}, requestDTO: {}", id, requestDTO);
+        ContentResult contentResult = new ContentResult();
+        HttpStatus httpStatus = HttpStatus.OK;
+        try {
+            Preconditions.checkNotNull(id);
+            Preconditions.checkNotNull(requestDTO);
+            Long userId = securityUtil.getAuthenticatedUserId();
+            //Ensure use logged in is same user who is updating the request to trade item
+            Preconditions.checkArgument(requestDTO.getTraderId().equals(userId));
+            Preconditions.checkArgument(id.equals(requestDTO.getId()));
+            RequestDTO updated  = this.requestService.update(requestDTO);
+            contentResult.setIsSuccess(true);
+            contentResult.setData(updated);
+            contentResult.setMessage(UserFriendlyMessage.REQUEST_UPDATE_SUCCEED);
+        } catch (BaseServiceException bse) {
+            logger.error("Error in updateRequest():", bse);
+            contentResult.setIsSuccess(false);
+            contentResult.setMessage(UserFriendlyMessage.REQUEST_UPDATE_FAILED);
+            httpStatus = HttpStatus.NOT_FOUND;
+        } catch (Exception e) {
+            logger.error("Error in updateRequest():", e);
+            contentResult.setIsSuccess(false);
+            contentResult.setMessage(UserFriendlyMessage.REQUEST_UPDATE_FAILED);
+            httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+        }
+        logger.info("End updateRequest");
+        return new ResponseEntity<>(contentResult, httpStatus);
+    }
+
+    /**
+     * Api to update the status of a request
+     *
+     * @param id id of the request to be updated.
+     * @param status new status of request
+     * @return Response to send to client.
+     */
+    @PutMapping("/{id}/{status}")
+    public ResponseEntity<ContentResult> updateRequestStatus(@PathVariable Long id, @PathVariable String status) {
+        logger.info("Start updateRequestStatus - id: {}, status: {}", id, status);
+        ContentResult contentResult = new ContentResult();
+        HttpStatus httpStatus = HttpStatus.OK;
+        try {
+            Preconditions.checkNotNull(id);
+            Preconditions.checkNotNull(status);
+            RequestDTO requestDTO = requestService.findById(id);
+            Long userId = securityUtil.getAuthenticatedUserId();
+
+            // Ensure user logged in is same user who is updating the request to trade item
+            Preconditions.checkArgument(requestDTO.getTraderId().equals(userId));
+
+            /* Update status of requestDTO */
+            requestDTO.setStatus(status);
+            RequestDTO updated  = this.requestService.update(requestDTO);
```

Comment:
Have to add checks to ensure that the status can be updated (might be better to create an updateStatus service layer method that deals with this)